src/test/java/com/aerospike/mapper/RecursiveObjectTest.java

```diff
@@ -0,0 +1,68 @@
+package com.aerospike.mapper;
+
+import org.junit.jupiter.api.Test;
+
+import com.aerospike.mapper.annotations.AerospikeKey;
+import com.aerospike.mapper.annotations.AerospikeRecord;
+import com.aerospike.mapper.tools.AeroMapper;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RecursiveObjectTest extends AeroMapperBaseTest {
+	@AerospikeRecord(namespace = "test", set = "A")
+	public static class A {
+		public String name;
+		public int age;
+		@AerospikeKey
+		public int id;
+//		@AerospikeReference(batchLoad = false)
+		public A a;
+		
+		A() {}
+
+		public A(String name, int age, int id) {
+			super();
+			this.name = name;
+			this.id = id;
+		}
+	}
+	
+	@AerospikeRecord(namespace = "test", set = "B")
+	public static class B {
+		@AerospikeKey
+		public int id ;
+		public A a;
+	}
+	
+	@Test
+	public void runTest() {
+		A a1 = new A("a", 10, 1);
+		a1.a = a1;
+		
+		AeroMapper mapper = new AeroMapper.Builder(client).build();
+		mapper.save(a1);
+		
+		A a2 = mapper.read(A.class, a1.id);
+		assertEquals(a1.age, a2.age);
+		assertEquals(a1.name, a2.name);
+		assertEquals(a1.id, a2.id);
+	}
```

Comment:
A slightly modified test doesn't pass
```java
@Test
public void runTest() {
	A a1 = new A("a", 10, 1);
	a1.a = new A("a2", 11, 11);

	AeroMapper mapper = new AeroMapper.Builder(client).build();
	mapper.save(a1);
	
	A a2 = mapper.read(A.class, a1.id);
	assertEquals(a1.age, a2.age);
	assertEquals(a1.name, a2.name);
	assertEquals(a1.id, a2.id);
	assertEquals(a1.a.id, a2.a.id);
}
```

Reply:
Actually, the test is flawed. The object mapper will only save the object being passed, not the entire object graph. You've created 2 objects, a1 and a1.a, but only saved a1. If we look at the definitions in Aerospike:

`
aql> select * from test.A
+-----+----+------+----+
| age | id | name | a  |
+-----+----+------+----+
| 10  | 1  | "a"  | 11 |
+-----+----+------+----+
1 rows in set (0.108 secs)
`

The second object, a1.a has not been saved. When the object mapper loads data it tries to load the subordinate object with an id of 11, but doesn't find this. So it sets the reference to be null.

If you change the save line to be 

`
mapper.save(a1, a1.a);
`

then the tests pass as the child object can now be found in Aerospike and loaded into the returned object hierarchy.